### PR TITLE
Quiet warnings about uninitialized values.

### PR DESCRIPTION
**Pull request by MaddTheSane**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [ ] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [ ] Other changes

### Description
This quiets warnings about uninitialized values, as well as one misused comma warning.

### Checklist

- [X] I have tested my changes locally and verified that they work as intended.
- [X] I have documented any new or modified functionality.
- [X] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [X] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### Additional Comments
The source codes were also ran through clang-format.

### CODE REVIEW

File: Descent3/gamesave.cpp [hunk 1]
```diff
@@ -622,7 +622,7 @@ bool LoadGameDialog() {
 
     fp = fopen(pathname, "rb");
     if (fp) {
-      int bm_handle;
+      int bm_handle = -1;
       int *pbm_handle;
```
Comment (Lgt2x) [inline Descent3/gamesave.cpp:626]: it's less important, but pointers should be initialized to `nullptr`

File: netcon/inetfile/inetgetfile.cpp [hunk 2]
```diff
@@ -248,7 +248,7 @@ BOOL InetGetFile::IsFileReceived() {
 }
 
 BOOL InetGetFile::IsFileError() {
-  int state;
+  int state = FTP_STATE_INTERNAL_ERROR;
```
Comment (Lgt2x) [inline netcon/inetfile/inetgetfile.cpp:251]: probably not totally right, but this whole function's logic needs rework

**Review by Lgt2x — APPROVED** [review]
Looks good!

File: Descent3/gamesave.cpp (hunk 1 above)
Comment (MaddTheSane) [reply Descent3/gamesave.cpp:626]: It didn't complain about `pbm_handle` being uninitialized. All the code paths that can initialize `pbm_handle` are handled by lines 629-633. If you're referencing the initializer for `pbm_handle` on line 632, I was just handling uninitialized values.

**Pull request merged**